lib/ansible/modules/windows/win_data_deduplication.ps1

```diff
@@ -0,0 +1,121 @@
+#!powershell
+
+# Copyright: 2019, rnsc(@rnsc) <github@rnsc.be>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+#AnsibleRequires -OSVersion 6.3
+
+$params = Parse-Args $args -supports_check_mode $true
+
+$drive_letter = Get-AnsibleParam -obj $params -name 'drive_letter' -type 'str' -failifempty $true
+$state = Get-AnsibleParam -obj $params -name 'state' -type 'str' -default 'present'
+$settings = Get-AnsibleParam -obj $params -name 'settings' -type 'dict' -default $null
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type 'bool' -default $false
+
+$result = @{
+  changed = $false
+  reboot_required = $false
+  msg = ""
+}
+
+function Set-DataDeduplication($volume, $state, $settings, $dedup_job) {
+
+  $current_state = 'absent'
+
+  try {
+    $dedup_info = Get-DedupVolume -Volume "$($volume.DriveLetter):"
+  } catch {
+    $dedup_info = $null
```

Comment:
When would this return `$null`, it seems like we have 3 states possible right now;

* `present`: - De duplication is enabled on the volume
* `absent`: Dedup is disabled on the volume
* `$null`: ? Possibly dedup is not set

Reply:
I got rid of this altogether.
It shouldn't be necessary to check for $null if the feature is installed.

Reply:
I got rid of this altogether.
It shouldn't be necessary to check for $null if the feature is installed.